scripts/main.js

```diff
@@ -31,18 +31,77 @@ var pins = [
     },
 ];
 
+var map;
 var mapContainer = document.querySelector(".map");
 
 var initMap = function () {
-    var map = new google.maps.Map(mapContainer, {
+    map = new google.maps.Map(mapContainer, {
         center: {lat: 33.7490, lng: -84.3880},
         zoom: 10
     });
     var infoWindow = new google.maps.InfoWindow();
-    addPinsToMap(map, infoWindow);
+    addMarkersToMap(infoWindow);
+    showVisibleMarkers();
 };
 
-var addPinsToMap = function (map, infoWindow) {
+var showVisibleMarkers = function () {
+    // Fired when the map becomes idle after panning or zooming.
+    google.maps.event.addListener(map, 'idle', function() {
+        // get bounds of current map viewport
+        var mapBounds = map.getBounds();
+        var searchListings = document.querySelector(".search-listings");
+        // loop through pins objects
+        pins.forEach(function (pin) {
+            // if mapBounds contains the pin position
+            if (mapBounds.contains(pin.position)) {
+                // display on screen
+                searchListings.appendChild(displayListing(pin));
+            }
+        })
+    });
+};
+
+var displayListing = function (pin) {
+    var listing = document.createElement("div");
+    listing.classList.add("listing");
+    listing.appendChild(listingImageDisplay(pin));
+    listing.appendChild(listingInfoDisplay(pin));
+    return listing;
+};
+
+var listingInfoDisplay = function (pin) {
+    var listingInfoContainer = document.createElement("div");
+    listingInfoContainer.classList.add("listing-info");
+    listingInfoContainer.appendChild(listingInfoTitle(pin));
+    listingInfoContainer.appendChild(listingInfoDescription(pin));
+    return listingInfoContainer;
+};
+
+var listingInfoTitle = function (pin) {
+    var listingTitle = document.createElement("p");
+    listingTitle.classList.add("listing-title");
+    listingTitle.textContent = pin.title;
+    return listingTitle;
+};
+
+var listingInfoDescription = function (pin) {
+    var listingDescription = document.createElement("p");
+    listingDescription.classList.add("listing-description");
+    listingDescription.textContent = pin.description;
+    return listingDescription;
+};
+
+var listingImageDisplay = function(pin) {
+    var imageContainer = document.createElement("div");
+    imageContainer.classList.add("listing-image");
+    var listingImage = document.createElement("img");
+    listingImage.classList.add("listing-image");
```

Comment:
Are both var imageContainer and var listingImage supposed to have the same class "listing-image"?

Reply:
I gave them the same class name for now so that the image itself fills up the entire div container it is located in.  Looking at it, I think we might be able to get rid of the image container all together.

Reply:
I think removing the image container would be fine, however have you tried seeing how the search listings would look like when expanded to different breakpoints (i.e. desktop screen size)? Would there be a possible need to have that image container later?